src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ParameterNameCheck.java

```diff
@@ -72,6 +78,12 @@
      */
     private boolean ignoreOverridden;
 
+    /** The visibility scope where methods are checked. */
+    private Scope scope = Scope.ANONINNER;
```

Comment:
Methods are checked if their access modifier is between `scope` (inclusive) and `excludeScope` (exclusive). The default is to "check all", so `scope` is set to the lowest visibility level and `excludeScope` to `null`, meaning "higher than public". 

The algorithm works because `NodeScope.isIn(Scope.ANONINNER)` returns `true` for all the visibility levels. I tried with `NodeScope.isIn(Scope.NONE)` but it does not. Besides I believe that saying in the documentation that the default scope is `NONE` to say that by default all the scopes are checked is a bit confusing.

Setting the default to `null` would work, of course, because then `matchScope` would check the specific value. Again, as a user I find a bit confusing that:
- `scope = null` means "all the visibility scopes"
- `excludeScope = null` means "no visibility scopes"

but if you prefer it this way, it's perfectly fine for me.